pkg/p2p/libp2p/libp2p.go

```diff
@@ -816,3 +832,23 @@ func (s *Service) Ready() {
 func (s *Service) Halt() {
 	close(s.halt)
 }
+
+func (s *Service) Ping(ctx context.Context, addr ma.Multiaddr) (rtt time.Duration, err error) {
+	ctx, cancel := context.WithCancel(ctx)
```

Comment:
not sure if this extra context is needed, since there is actually no cases where it is being cancelled. on shutdown, the hive will cancel the context which is propagated to this call, so probably this is not needed